src/main/java/com/hrznstudio/titanium/client/gui/addon/ItemstackFilterGuiAddon.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * This file is part of Titanium
+ * Copyright (C) 2019, Horizon Studio <contact@hrznstudio.com>.
+ *
+ * This code is licensed under GNU Lesser General Public License v3.0, the full license text can be found in LICENSE.txt
+ */
+
+package com.hrznstudio.titanium.client.gui.addon;
+
+import com.hrznstudio.titanium.api.client.AssetTypes;
+import com.hrznstudio.titanium.api.filter.FilterSlot;
+import com.hrznstudio.titanium.client.gui.addon.interfaces.IClickable;
+import com.hrznstudio.titanium.client.gui.addon.interfaces.INetworkable;
+import com.hrznstudio.titanium.client.gui.asset.IAssetProvider;
+import com.hrznstudio.titanium.filter.ItemstackFilter;
+import com.hrznstudio.titanium.util.AssetUtil;
+import com.mojang.blaze3d.platform.GlStateManager;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.AbstractGui;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.inventory.ContainerScreen;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+
+import java.awt.*;
+import java.util.Objects;
+
+public class ItemstackFilterGuiAddon extends BasicGuiAddon implements IClickable {
+
+    private final ItemstackFilter filter;
+
+    public ItemstackFilterGuiAddon(ItemstackFilter filter) {
+        super(filter.getFilter()[0].getX(), filter.getFilter()[0].getY());
```

Comment:
Why is this based on the first slot?

Reply:
Because the first slot is the 0, 0 coords of the filter